File: core/src/remove/tanks/game/level/preload/PreloadData.java
package remove.tanks.game.level.preload;

import java.util.List;

/**
 * @author Mateusz Długosz
 */
public final class PreloadData {
    private final List<String> entityPrefabCodes;
    private final List<String> assetIds;

    public PreloadData(List<String> entityPrefabCodes, List<String> assetIds) {
        this.entityPrefabCodes = entityPrefabCodes;
        this.assetIds = assetIds;
    }

    public List<String> getEntityPrefabCodes() {
        return entityPrefabCodes;
    }

    public List<String> getAssetIds() {
        return assetIds;
    }
}

File: tools/docs-generator/lib/level/preload/preload_data.py
class PreloadData(object):
    def __init__(self, entity_prefab_codes=None, asset_ids=None):
        if asset_ids is None:
            asset_ids = []
        if entity_prefab_codes is None:
            entity_prefab_codes = []

        self.entity_prefab_codes = entity_prefab_codes
        self.asset_ids = asset_ids

    def get_entity_prefab_codes(self):
        return self.entity_prefab_codes

    def get_asset_ids(self):
        return self.asset_ids

    def __str__(self):
        return "PreloadData(entity_prefab_codes=[{}], asset_ids=[{}])" \
            .format(
                ", ".join('\'{}\''.format(val) for val in self.entity_prefab_codes),
                ", ".join('\'{}\''.format(val) for val in self.asset_ids)
            )
